### PR TITLE
Fixing cat feedback

### PR DESCRIPTION
**Pull request by amontenegro**

### CODE REVIEW

**Comment by amontenegro** [comment]
https://trello.com/c/m4iF9o6b/1932-when-creating-a-record-the-credit-name-field-is-confused-about-it-s-visibility-setting-if-you-don-t-specify-it

File: orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java [hunk 1]
```diff
@@ -2220,32 +2220,32 @@ private void setDefaultVisibility(OrcidProfile orcidProfile) {
                 OrcidBio bio = orcidProfile.getOrcidBio();
                 if (bio.getBiography() != null) {
                     if (bio.getBiography().getVisibility() == null) {
-                        bio.getBiography().setVisibility(OrcidVisibilityDefaults.BIOGRAPHY_DEFAULT.getVisibility());
+                        bio.getBiography().setVisibility(Visibility.PRIVATE);
```
Comment (rcpeters) [inline orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java]: @lpaglione @amontenegro @caoyler is this right? Bio is always going to be set to private when one is created?

**Comment by amontenegro** [comment]
https://trello.com/c/m4iF9o6b/1932-when-creating-a-record-the-credit-name-field-is-confused-about-it-s-visibility-setting-if-you-don-t-specify-it

If I get understood well the last comment from @caoyler if the bio doesnt have any visibility specified, it should be private, the same for all other fields.

Is that ok @caoyler ?

**Comment by rcpeters** [comment]
@caoyler @amontenegro We are going to default public to default private for all of bio, including bio/bio Also why isn't this change coded in OrcidVisibilityDefaults?

**Comment by amontenegro** [comment]
Hey @rcpeters, 

I didnt change it directly in the OrcidVisibilityDefaults, because those values are used in many other places, so, we will default all of them to private, while, with this change we just want to change it for new profiles.

Here is the classes that uses the OrcidVisibilityDefaults constants: 

BIOGRAPHY_DEFAULT 	/orcid-core/src/main/java/org/orcid/pojo/ajaxForm/BiographyForm.java	
BIOGRAPHY_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/forms/ChangePersonalInfoForm.java
COUNTRY_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/forms/ChangePersonalInfoForm.java
COUNTRY_DEFAULT 	/orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidJaxbCopyManagerImpl.java
CREDIT_NAME_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/forms/ChangePersonalInfoForm.java
CREDIT_NAME_DEFAULT 	/orcid-core/src/main/java/org/orcid/pojo/ajaxForm/Contributor.java
CREDIT_NAME_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/controllers/FundingsController.java
CREDIT_NAME_DEFAULT 	/orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidJaxbCopyManagerImpl.java
CREDIT_NAME_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/controllers/WorksController.java
EXTERNAL_IDENTIFIER_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/forms/ChangePersonalInfoForm.java
EXTERNAL_IDENTIFIER_DEFAULT 	/orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidJaxbCopyManagerImpl.java
KEYWORD_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/forms/ChangePersonalInfoForm.java
KEYWORD_DEFAULT 	/orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidJaxbCopyManagerImpl.java
RESEARCHER_URLS_DEFAULT 	/orcid-core/src/main/java/org/orcid/core/manager/impl/AdminManagerImpl.java
RESEARCHER_URLS_DEFAULT 	/orcid-web/src/main/java/org/orcid/frontend/web/forms/ChangePersonalInfoForm.java
RESEARCHER_URLS_DEFAULT 	/orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidJaxbCopyManagerImpl.java

So, maybe we can do a OrcidRegistrationVisibilityDefaults with null values and use it for this change, or, if you are ok, we can change the OrcidVisibilityDefaults and default everything to private in those classes to.

Let me know how to proceed.

Thanks

**Comment by caoyler** [comment]
When creating a record via the API, any field without a visibility setting
should be private. Any field with a visibility attribute should use the
specified visibility.

-Cat

On Wed, Apr 15, 2015 at 10:59 AM, Angel Montenegro <notifications@github.com

**Comment by amontenegro** [comment]
Hey @rcpeters, 

Please check this, so, now, the privacy will depend on the method used to create the profile, if it is created by a member (Bulk or by the API), all the sections that have not specified a visibility will be set to OrcidVisibilityDefaults.CREATED_BY_MEMBER_DEFAULT which is private, but, if the user is created in the UI, the default value will be taken from the corresponding variable in OrcidVisibilityDefaults.

Thanks

**Pull request merged**